### ISSUE TITLE
🦂 [Bug]Stratholme incorrect things / polish / bugs

### ISSUE THREAD

**Issue by ShiyoKozuki** [vmangos/core#2402]
## 🦂 Bug report - Dungeon
Lots wrong, see below for how it should be. Mainly the end with Rammstein / Baron.
Thuzadin necromancer drain life also drains mana, scholo ones that mind flay also do. Unsure if accurate.
Boxes with holy water is super buggy - >
https://github.com/vmangos/core/issues/2398

Double chest spawn here once:
![image](https://github.com/vmangos/core/assets/58101246/81fb609b-bcaa-482e-b563-37b4070fd5c8)

### Expected behavior

https://youtu.be/aBAKPCzVs-g?t=3539
Mindless Undead Skeletons are supposed to run at the group in a congo line after you kill Rammstein

https://youtu.be/aBAKPCzVs-g?t=3594
Black guard sentries aren't supposed to spawn until after the non-elites are all killed. They are also supposed to all force link together.

https://youtu.be/RMdgfp2x95c?t=3742
Baron Rivendare shouldn't aggro you from as far as he does so you can enter the room and run to the back letting everyone enter before engaging him and him closing the door.

### Version & Environment

Client Version:  1.12.1

Commit Hash:   newest

OS Client:  win10
OS Server:  win10

### Crashlog

- None

**Comment by GoonBoiBarry** [comment]
> Black guard sentries aren't supposed to spawn until after the non-elites are all killed. They are also supposed to all force link together.

I don't think this is a bug. The video you show has the non-elites killed over a pretty long duration and still have to wait for the black guards to come out about 20 seconds.

https://youtu.be/xIq0SXsLUtc?t=3121
This video shows a group kill the non-elites much faster and have to wait about 1 minute and 10 seconds for the black guards to come out.

The black guards coming out appears to be on a timer after Ramstein dies. Both videos show that approximately 1 minute and 30 seconds after ramstein dies, the black guards come out regardless of when the non-elites were killed.

Here are the timestamps to show that:
https://youtu.be/aBAKPCzVs-g?t=3528 Ramstein death 58:48
https://youtu.be/aBAKPCzVs-g?t=3621 Black guards 1:00:21

https://youtu.be/xIq0SXsLUtc?t=3114 Ramstein death 51:54
https://youtu.be/xIq0SXsLUtc?t=3206 Black guards 53:26
